### PR TITLE
Fixed issue w/ git commit, added new files for this week.

### PR DESCRIPTION
**Pull request by vanulor**
Refactored report, and send reports, added more functions.  Added comprehensions and exceptions.  Future work will include reworking the thank you report because it is a bit unwieldy.

### CODE REVIEW

File: students/BrianNisonger/session4/trigram.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+def make_story(tris,new_words,story):
+    new_list=new_words.split()
+    new_tuple=tuple(new_list)
+    if new_tuple in tris:
+        next_word=str(tris[new_tuple])
+        punct=[",",".","-","(",")","\n","[","]","'"]   
+        for punc in punct:
+            next_word=next_word.replace(punc,"")
+        new_words=f'{new_tuple[1]} {next_word}'
```
Comment (PythonCHB) [inline students/BrianNisonger/session4/trigram.py]: building a long string up by adding a to it bit by bit is pretty inefficient -- better to keep it all in a list of works, and join then all in the end.

That will save you having to split() the words up as well.

File: students/BrianNisonger/session5/except_exercise.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+# parameters. (pun intended)
+
+langs = ['java', 'c', 'python']
+try:
+    more_joke = more_fun(langs[0])
+except IndexError:
+    more_joke=more_fun(langs[1])
+finally:    
+    last_fun()
+
```
Comment (PythonCHB) [inline students/BrianNisonger/session5/except_exercise.py]: perfect!

**Comment by PythonCHB** [comment]
you've got conflicts to resolve.

I"m having trouble figureing out how / why this happens, but try:

git pull upstream master

then you may have a conflict in your machine -- if so, resolve it.

then

git commit

git push

OR -- resolve it in the gitHub GUI on this issue.

**Comment by vanulor** [comment]
Fixed issues added new assignments

**Comment by PythonCHB** [comment]
Merging this -- do a new PR when you have html_render to review.

**Pull request merged**